tokio-trace/tokio-trace-core/src/metadata.rs

```diff
@@ -205,6 +227,11 @@ impl<'a> Metadata<'a> {
     pub fn callsite(&self) -> callsite::Identifier {
         self.fields.callsite()
     }
+
+    /// Returns the kind of callsite.
+    pub fn callsite_kind(&self) -> &CallsiteKind {
```

Comment:
Again, if `CallsiteKind` is going to be an enum, it's probably best not to expose it publicly. We would probably want to replace this with `Metadata::is_span` and `is_event` functions instead (and/or replace the public `CallsiteKind` type with a struct).